File: Z.ExtensionMethods-master/src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.Int64/GenericInt/Int64.FactorOf.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library


Public Module Extensions_300

	''' <summary>
	'''     An Int64 extension method that factor of.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="factorNumer">The factor numer.</param>
	''' <returns>true if it succeeds, false if it fails.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function FactorOf(this As Int64, factorNumer As Int64) As Boolean
		Return factorNumer Mod this = 0
	End Function
End Module

File: Z.ExtensionMethods-master/src/Z.Core/System.Int64/GenericInt/Int64.FactorOf.cs
using System;

public static partial class Extensions
{
    /// <summary>
    ///     An Int64 extension method that checks if the value is a factor of the given factor number.
    /// </summary>
    /// <param name="this">The value to act on.</param>
    /// <param name="factorNumer">The factor number.</param>
    /// <returns>true if the value is a factor of the given factor number, false if not.</returns>
    public static bool FactorOf(this long @this, long factorNumer)
    {
        return factorNumer % @this == 0;
    }
}
